### ISSUE TITLE
play.api.UnexpectedException: Unexpected exception[NoSuchMethodError: play.api.ApplicationLoader$.createContext(Lplay/api/Environment;Lscala/collection/immutable/Map;Lscala/Option;Lplay/core/WebCommands;Lplay/api/inject/DefaultApplicationLifecycle;)Lplay/api/ApplicationLoader$Context;]

### ISSUE THREAD

**Issue by josefmaxa** [play2-maven-plugin/play2-maven-test-projects#3]
In my app play2:run throws this Exception? Have you ever come across that?
play.api.UnexpectedException: Unexpected exception[NoSuchMethodError: play.api.ApplicationLoader$.createContext(Lplay/api/Environment;Lscala/collection/immutable/Map;Lscala/Option;Lplay/core/WebCommands;Lplay/api/inject/DefaultApplicationLifecycle;)Lplay/api/ApplicationLoader$Context;]

_Note_ the example's pom refers to 1.0.0-rc3-SNAPSHOT. I had to change that to 1.0.0-rc2 to make it work._

Using: 
<properties>
        <dozer.version>6.1.0</dozer.version>

        <play.version>2.6.12</play.version>
        <play-enhancer.version>1.2.2</play-enhancer.version>
        <deadbolt-version>2.6.4</deadbolt-version>

        
        <scala.version>2.12.4</scala.version>

        <jar.plugin.version>2.6</jar.plugin.version>

        <play2.plugin.version>1.0.0-rc2</play2.plugin.version>
        <sbt-compiler.plugin.version>1.0.0</sbt-compiler.plugin.version>
        <sbtrun.plugin.version>1.0.1</sbtrun.plugin.version>

        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>

    </properties>

In working build.sbt we have Akka disabled and Netty enabled. Could it be the issue?:
lazy val root = (project in file(".")).enablePlugins(PlayJava, SbtWeb, PlayScala, PlayNettyServer)
                                               .disablePlugins(PlayAkkaHttpServer)

Log server run:
p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000

**Comment by josefmaxa** [comment]
Thanks for the second pair of eyes.
Here is how I resolved the issue.

I had dependency on play-netty-server_2.12 in my pom, but I'm using deadbolt-java_2.12 which pulls play-akka-http-server_2.12.

Now the spooky thing. 

- If the dependency on play-netty-server_2.12 is **above** deadbolt-java_2.12 everything works nicely (play-akka-http-server_2.12 doesn't need to be excluded from deadbolt-java_2.12).  Attached **pom.working.xml**
- If the dependency on play-netty-server_2.12 is **bellow** deadbolt-java_2.12 you will get the issue I described (play-akka-http-server_2.12 excluding deadbolt-java_2.12 produces a different error). Attached **pom.broken.xml**

These poms are modified **com.google.code.play2-maven-plugin.test-projects.play26.java**
[pom.zip](https://github.com/play2-maven-plugin/play2-maven-test-projects/files/1815808/pom.zip)

**Comment by gslowikowski** [comment]
Exclude `play-akka-http-server_2.12` from `deadbolt-java_2.12`:
```xml
        <dependency>
            <groupId>be.objectify</groupId>
            <artifactId>deadbolt-java_2.12</artifactId>
            <version>${deadbolt-version}</version>
            <exclusions>
                <exclusion>
                    <groupId>com.typesafe.play</groupId>
                    <artifactId>play-akka-http-server_2.12</artifactId>
                </exclusion>
            </exclusions>
       </dependency>
```

In my opinion, deadbolt should not pull `play-akka-http-server` dependency transitively.

**Comment by josefmaxa** [comment]
Yes I have play-akka-http-server_2.12 excluded from deadbolt-java_2.12, but as I wrote, it has no effect on this issue.
